target/scripts/startup/setup.d/mail_state.sh

```diff
@@ -80,13 +80,11 @@ function _setup_save_states
     chown -R postfix:root /var/mail-state/spool-postfix
     chown root:root /var/mail-state/spool-postfix
     # These two require the postdrop(103) group:
-    chgrp -R postdrop /var/mail-state/spool-postfix/maildrop
-    chgrp -R postdrop /var/mail-state/spool-postfix/public
-    # These all have root ownership at the src location:
-    chown -R root /var/mail-state/spool-postfix/dev
-    chown -R root /var/mail-state/spool-postfix/etc
-    chown -R root /var/mail-state/spool-postfix/lib
-    chown -R root /var/mail-state/spool-postfix/pid
-    chown -R root /var/mail-state/spool-postfix/usr
+    chgrp -R postdrop /var/mail-state/spool-postfix/{maildrop,public}
+  elif [[ ${ONE_DIR} -eq 1 ]]
+  then
+    _log 'warn' "'ONE_DIR=1' but no volume was mounted to '${STATEDIR}'"
+  else
+    _log 'debug' 'Not consolidating state (because it has been disabled)'
```

Comment:
Did I miss something here? `ONE_DIR` is used above in conjunction with `&& [[ -d ${STATE_DIR} ]]` 🙂

Reply:
> Did I miss something here?

The warning would always be output with `variables.sh` setting `ONE_DIR=1` as default? (_unless there is an explicit opt-out with `ONE_DIR=0`_).

**EDIT:** For UX you can ignore the below content. Your decision here is the right way to go :+1: (_opt-out of warning via explicit `ONE_DIR=0`_)

---

I think I had another comment in this thread that didn't seem to get added during review :grimacing: 

I was proposing `variables.sh` to set `ONE_DIR=1` if `/var/mail-state` existed, so that the warning here would not show up by default.

- The test suite doesn't run with a `/var/mail-state` volume in any tests, so we technically use `ONE_DIR=0`.
- It used to be opt-in ENV until the default was changed. Since we expect it at `/var/mail-state` anyway, we could just derive it that way.
- `ONE_DIR` is not used anywhere else in our scripts, just special handling in `mail_state.sh`.
- The warning then becomes redundant, opt-in via existence of `/var/mail-state`.

Then again, if there is no explicit opt-out with `ONE_DIR=0`, for new users especially it probably does make sense to display a warning :+1:

Reply:
The error message is not triggerd, when `ENABLE_SRS=1` and `ONE_DIR=1` is set and no volume is mounted, because the directory is created in the setup process here:

https://github.com/docker-mailserver/docker-mailserver/blob/4e9ffbf2241a860f6a4d82a97c63fa82d8722fbc/target/scripts/startup/setup.d/postfix.sh#L173

Reply:
Any idea how to solve that?

Reply:
SRS seems pretty broken to me. For example, SRS stores stuff in `${POSTSRSD_STATE_DIR}`, however there is no symlink pointing to it..
I will take a closer look later.

Reply:
Fixed in https://github.com/docker-mailserver/docker-mailserver/pull/3158